docs/source/guide/configuration.rst

```diff
@@ -17,100 +20,319 @@ default region::
+(``~/.aws/credentials``).  By using the shared credentials file, you can use a
+single file for credentials that will work in all the AWS SDKs.
+
+
+Configuration
+=============
+
+In addition to credentials, you can also configure non-credential values.  In
+general, boto3 follows the same approach used in credential lookup: try various
+locations until a value is found.  Boto3 uses these sources for configuration:
+
+* Explicitly passed as the ``config`` paramter when creating a client.
+* Environment variables
+* The ``~/.aws/config`` file.
+
+Environment Variable Configuration
+----------------------------------
+
+``AWS_ACCESS_KEY_ID``
+    The access key for your AWS account.
+
+``AWS_SECRET_ACCESS_KEY``
+    The secret key for your AWS account.
+
+``AWS_SESSION_TOKEN``
+    The session key for your AWS account.  This is only needed when
+    you are using temporary credentials.  The ``AWS_SECURITY_TOKEN``
+    environment variable can also be used, but is only supported
+    for backwards compatibility purposes.  ``AWS_SESSION_TOKEN`` is
+    supported by multiple AWS SDKs besides python.
+
+``AWS_DEFAULT_REGION``
+    The default region to use, e.g. ``us-west-2``, ``us-west-2``, etc.
+
+``AWS_PROFILE``
+    The default profile to use, if any.  If no value is specified, boto3
+    will attempt to seach the shared credentials file and the config file
+    for the ``default`` profile.
+
+``AWS_CONFIG_FILE``
+    The location of the config file used by boto3.  By default this
+    value is ``~/.aws/config``.  You only need to set this variable if
+    you want to change this location.
+
+``AWS_SHARED_CREDENTIALS_FILE``
+    The location of the shared credentials file.  By default this value
+    is ``~/.aws/credentials``.  You only need to set this variable if
+    you want to change this location.
+
+``AWS_CA_BUNDLE``
+    The path to a custom certificate bundle to use when establishing
+    SSL/TLS connections.  Boto3 includes a bundled CA bundle it will
+    use by default, but you can set this environment variable to use
+    a different CA bundle.
+
+``AWS_METADATA_SERVICE_TIMEOUT``
+    The number of seconds before a connection to the instance metadata
+    service should time out.  When attempting to retrieve credentials
+    on an EC2 instance that has been configured with an IAM role,
+    a connection to the instance metadata service will time out after
+    1 second by default.  If you know you are running on an EC2 instance
+    with an IAM role configured, you can increase this value if needed.
+
+``AWS_METADATA_SERVICE_NUM_ATTEMPTS``
+    When attempting to retrieve credentials on an EC2 instance that has
+    been configured with an IAM role, boto3 will only make one attempt
+    to retrieve credentials from the instance metadata service before
+    giving up.  If you know your code will be running on an EC2 instance,
+    you can increase this value to make boto3 retry multiple times
+    before giving up.
+
```

Comment:
We should, I omitted it because I wasn't sure exactly what to say about it, as it's not something you typically need to mess with.  But we should document, I'll just need to figure out how to word it.